Run pyupgrade on tools/

foolip opened the pull request:
Command used, with pyupgrade 2.31.1 installed:

> find tools -name '*.py' | grep -v third_party | xargs pyupgrade --keep-percent-format --py36-plus

foolip commented on tools/ci/tc/tests/test_valid.py:
```diff
@@ -2,7 +2,7 @@
 
 import jsone
-import mock
+from unittest import mock
```
I don't like the order of this. @gsnedders @jgraham are there tools to fix/enforce import ordering in Python?

foolip commented:
I've also tried running `pyupgrade` on all code outside of tools/, but ran into what seems like a bug: https://github.com/asottile/pyupgrade/issues/461

jgraham commented:
Import ordering in Python isn't easy to order in a generic way because imports have side effects. So in-place replacement is the safe thing to do in this case. If we want to fix it we'll probably need to do it by hand (I think there are tools but they aren't safe for that reason). 

FWIW the typical style in the absence of constraints is to put `from foo import bar` imports below all plain `import baz` imports in the same category (stdlib/other libraries/local).

foolip commented:
https://github.com/asottile/pyupgrade/issues/461 was quickly fixed and released as 2.19.2, and with that running pyupgrade on all of our code (except tools/third_party/) works.

gsnedders commented on tools/wptserve/tests/functional/test_request.py:
```diff
@@ -125,13 +124,13 @@ def handler(request, response):
 
         # Try some non-ASCII characters and the server shouldn't crash.
-        encoded_text = u"你好".encode("utf-8")
+        encoded_text = "你好".encode()
```
Unconvinced by this, hmm?

foolip replied:
It's equivalent per https://docs.python.org/3/library/stdtypes.html#str.encode, but maybe you mean being explicit is better?

foolip replied:
I'll revert all changes like this.

foolip committed:
Put back .encode("utf-8")

foolip commented:
I've fixed up import ordering now, but then realized that we're using https://pypi.org/project/mock/ and most likely don't need to. It would be best to make that change separately however.

foolip commented:
I've broken that out into https://github.com/web-platform-tests/wpt/pull/29323 and reverted changes around `mock`.

gsnedders commented:
> Command used:
> 
> > find tools -name '*.py' | grep -v third_party | xargs pyupgrade --keep-percent-format --py36-plus

Are we sure we want `--keep-percent-format`? Especially with pyupgrade converting simple cases to f-strings.

> I don't like the order of this. @gsnedders @jgraham are there tools to fix/enforce import ordering in Python?

isort is by far the most common tool, though as @jgraham mentions it is a semantic change. We don't do much monkey-patching so I expect automated tools will work okay for us?

--

Also: I do wonder if we're going to do something like this whether we should also try and do a bunch of other things around the same time (like `isort` and reformatting everything with `black`, moving `typing` imports out from `if MYPY`)? If we're going to do a bunch and make merge conflicts happen, we should probably try and get as much of this done in one go as possible, and ideally provide tooling to merge older work.

jgraham commented:
> Are we sure we want --keep-percent-format? Especially with pyupgrade converting simple cases to f-strings.

I think I'm happier with keeping it for now. The `.format` thing always seemed awful, and s-strings are probably the best solution, but `%` formatting seems fine. Or at least worth breaking out into a different change.

 > isort is by far the most common tool, though as @jgraham mentions it is a semantic change. We don't do much monkey-patching so I expect automated tools will work okay for us?

With the localpaths stuff in particular, I totally don't guarantee that blindly running isort is a good idea. We used it on treeherder at some point and I generally found it to be way more trouble than it's worth.

 > Also: I do wonder if we're going to do something like this whether we should also try and do a bunch of other things around the same time (like isort and reformatting everything with black, moving typing imports out from if MYPY)?

I understand the appeal of auto-formatting code, and like using Rustfmt, but I honestly find that Black actively makes code harder to read (mostly because it has a tendency to format very similar constructs in totally different ways according to the precise line length). Plus if we want to insist that code remains "correctly" formatted at all times we need to agree a precise version and ensure that people can use it in vendor repos, otherwise there are going to be CI failures every time someone tries to upstream a Python PR. 

Overall I don't think it's worth the effort to try to use these tools in wpt.

foolip commented:
I went with `--keep-percent-format` not because I have an opinion about `%` vs. `.format()` vs. f-strings, but just to make the changes smaller. With hindsight, I should have used `--keep-mock` to avoid those changes too. Too late for `--keep-runtime-typing`, whatever that did.

foolip commented on tools/wptrunner/wptrunner/executors/base.py:
```diff
@@ -677,7 +677,7 @@ def run(self):
         try:
             self.result = True, self.func(self.session, self.path, self.timeout)
-        except (socket.timeout, IOError):
+        except (socket.timeout, OSError):
```
Looks like this and all `socket.timeout` cases can be simplified since per https://docs.python.org/3/library/socket.html#socket.timeout it's a subclass of `OSError` now.

foolip replied:
I've filed https://github.com/asottile/pyupgrade/issues/471 about this.

foolip commented:
I've broken out the `socket.error` and `socket.timeout` changes into https://github.com/web-platform-tests/wpt/pull/29362 and would like to land that first.

foolip committed:
Run pyupgrade on tools/

Command used, with pyupgrade 2.31.1 installed:

> find tools -name '*.py' | grep -v third_party | xargs pyupgrade --keep-percent-format --py36-plus

foolip committed:
Fix flake8 errors

foolip commented:
Looks like I just forgot this after breaking out some bits. Now recreated with the same fixes applied, so landing.
